src/gateway/http.ts

```diff
@@ -96,21 +97,13 @@ export class HTTP extends Gateway {
         httpOptions.onRequestSocketAssigned(requestParams)
       }
 
-      socket.on('lookup', () => {
-        if (httpOptions.onSocketLookup) {
-          httpOptions.onSocketLookup(requestParams)
-        }
-      })
-      socket.on('connect', () => {
-        if (httpOptions.onSocketConnect) {
-          httpOptions.onSocketConnect(requestParams)
-        }
-      })
-      socket.on('secureConnect', () => {
-        if (httpOptions.onSocketSecureConnect) {
-          httpOptions.onSocketSecureConnect(requestParams)
-        }
-      })
+      if (httpRequest.reusedSocket) {
+        return
+      }
+
+      this.assignSocketListener(socket, 'lookup', httpOptions.onSocketLookup?.bind(null, requestParams))
+      this.assignSocketListener(socket, 'connect', httpOptions.onSocketConnect?.bind(null, requestParams))
+      this.assignSocketListener(socket, 'secureConnect', httpOptions.onSocketSecureConnect?.bind(null, requestParams))
```

Comment:
Not sure I like the indirection of having a utility function here. I can see the appeal for making these one-liners, but the rest of the code base doesn't really resonate well with it.
- we do not use `.bind(null`, which purpose is to set `this` to not be the this of the caller but to be "stateless". But then why even put it as a class function (its not even private)
- the `bind` is also a bit tricky since it is returning a new function, I would prefer to avoid it, it's too clever for me :)

I think it is much more clear / in line with the rest of the code base if we do this:

```suggestion
      if (httpOptions.onSocketLookup) {
        socket.on('lookup', () => {
          httpOptions.onSocketLookup?.(requestParams)
        })
      }
      if (httpOptions.onSocketConnect) {
        socket.on('connect', () => {
          httpOptions.onSocketConnect?.(requestParams)
        })
      }
      if (httpOptions.onSocketSecureConnect) {
        socket.on('secureConnect', () => {
          httpOptions.onSocketSecureConnect?.(requestParams)
        })
      }
```

Even if it is more "verbose" I think it is more clear what is going on, WDYT?

Reply:
Sounds good to me, I agree that the bind approach is a bit too much 😄